### ISSUE TITLE
Develop the reschedule feature for timeline 

### ISSUE THREAD

**Issue by Tushar-4781** [tijlleenders/ZinZen#1400]
Reschedule implementation could be as simple as providing a `notOn` for the required 'push' period.  
So reschedule could show you an list of reschedule/postpone options:  
- 1 hour
- 3 hours
- 1 day
- 3 days
- 7 days
- 30 days

If the Goal has a deadline, only show the postpone options that leave at least the duration of the goal between the new start and the deadline.

All the scheduler needs is an extra `notOn` field (separate from filters/timeBudget) for the start of the calendar till the current scheduled start time + postpone time. This field contains **an array** of start/end tuples that represent hours it should not be scheduled on:

```
  {
    "startDate": "2022-01-01T00:00:00",
    "endDate": "2022-01-02T00:00:00",
    "goals": [
      {
        "id": "1",
        "title": "shopping",
        "minDuration": 1,
        "start": "2022-01-01T10:00:00",
        "deadline": "2022-01-01T13:00:00",
        "notOn": [
          {
            "start": "2022-01-01T00:00:00",
            "end": "2022-01-01T11:00:00"
          }
        ]
      }
    ]
  }
```

**Comment by Tushar-4781** [comment]
Blocked by Rust Scheduler

**Comment by vinaybadgujar102** [comment]
@Tushar-4781 what is the purpose of dumpBox collection

**Comment by Tushar-4781** [comment]
it is to save the output of scheduler that's it for now

**Comment by tijlleenders** [comment]
I think this feature is worth including a test for (using Playwright since we're not using Cypress anymore).

**Comment by tijlleenders** [comment]
Let Something with start date X.
Postpone by Y hours means the notOn should be [start date time X, start date time X + Y[
